src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -25,30 +23,30 @@ public ProductService(MessagesDTO dto) {
     }
 
     // get supported components defined in vip service
-    public JSONArray getComponentsFromRemoteVIP() {
+    public List<String> getComponents() {
         BaseDTO baseDTO = new BaseDTO();
         baseDTO.setProductID(dto.getProductID());
         baseDTO.setVersion(dto.getVersion());
         ProductBasedOpt dao = new ProductBasedOpt(baseDTO);
-        return dao.getComponentsFromRemoteVIP();
+        return dao.getComponents();
     }
 
     // get supported locales defined in vip service
-    public JSONArray getSupportedLocalesFromRemoteVIP() {
+    public List<String> getSupportedLocales() {
         BaseDTO baseDTO = new BaseDTO();
         baseDTO.setProductID(dto.getProductID());
         baseDTO.setVersion(dto.getVersion());
         ProductBasedOpt dao = new ProductBasedOpt(baseDTO);
-        return dao.getSupportedLocalesFromRemoteVIP();
+        return dao.getSupportedLocales();
     }
 
     public List<Map> getAllComponentTranslation() {
```

Comment:
Because this method is in ProductService. How about put all translation methods into one service?